### PR TITLE
Adding documentation for module options

### PR DESCRIPTION
**Pull request by hannes-harnisch**
**What does this PR do?**

The website currently does not fully document the module-related options for the `compileas` setting. I implemented these settings in #1570. I added basic descriptions about what the individual settings do. The terms used reflect those used in the C++20 standard. I think more detailed descriptions aren't necessary, as they would probably easily branch out into a lengthy description of C++20 modules, which shouldn't be the purpose of this page.

**How does this PR change Premake's behavior?**

In no way. It's purely documentation.

**Anything else we should know?**

No.

**Did you check all the boxes?**

- [X] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [X] Add unit tests showing fix or feature works; all tests pass
- [X] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [X] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] Minimize the number of commits
- [X] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

**Review by samsinsane — APPROVED** [review]

File: website/docs/compileas.md [hunk 1]
```diff
@@ -12,9 +12,9 @@ compileas "value"
 * `C++` - Compile as a C++ source file.
 * `Objective-C` - Compile as an Objective-C source file.
 * `Objective-C++` - Compile as an Objective-C++ source file.
-* `Module` - Needs documentation
-* `ModulePartition` - Needs documentation
-* `HeaderUnit` - Needs documentation
+* `Module` - Compile as a C++20 module interface unit.
+* `ModulePartition` - Compile as a C++20 module interface partition.
+* `HeaderUnit` - Compile as a C++20 header unit.
```
Comment (Jarod42) [inline website/docs/compileas.md:17]: Shouldn't below "Availability"'s section should states that those parameter are only supported by msvc generator?

Comment (hannes-harnisch) [reply website/docs/compileas.md:17]: Amended.

**Comment by hannes-harnisch** [comment]
I'm unsure on what makes the website check fail.

**Comment by KyrietS** [comment]
I'm looking into this

**Comment by KyrietS** [comment]
Thanks for your help in documenting those values! 🤗

**Pull request merged**